src/extension.ts

```diff
@@ -81,14 +81,21 @@ export async function activate(context: vscode.ExtensionContext) {
     }),
 
     vscode.commands.registerCommand('RedisForVSCode.addDatabase', (args) => {
-      WebviewPanel.getInstance({
+      const panel = WebviewPanel.getInstance({
         context,
         route: 'main/add_database',
         title: vscode.l10n.t('Redis for VS Code - Add Database connection'),
         viewId: ViewId.AddDatabase,
         handleMessage: (message) => handleMessage(message),
         message: args,
       })
+
+      if (args?.ssoFlow && args?.source) {
+        panel.postMessage({
+          action: 'OpenOAuthSsoDialog',
+          data: { source: args?.source, ssoFlow: args?.ssoFlow },
+        })
+      }
```

Comment:
If we have the check logic in the `case VscodeMessageAction.OpenOAuthSsoDialog` in `webviews/src/index`, lets remove it from here. I know that we offline talked about single responsibility and how we don't want to make the other module "smart" about this logic, but if we have it there, let's not check it here.